compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5359,6 +5359,15 @@ public void visit(BLangBinaryExpr binaryExpr, AnalyzerData data) {
                 if (leftConstituent != null && rightConstituent != null) {
                     actualType = new BXMLType(BUnionType.create(null, leftConstituent, rightConstituent), null);
                     break;
+                } else if (leftConstituent != null || rightConstituent != null) {
+                    if (leftConstituent != null && types.isAssignable(rhsType, symTable.stringType)) {
+                        actualType = new BXMLType(BUnionType.create(null, leftConstituent, symTable.xmlTextType), null);
```

Comment:
IMO, creating this kind of union type for the actual type is not correct. For example, consider the following case: it creates a union type with the same type twice, which is incorrect.
```ballerina
type Foo xml<xml:Element|xml:Comment>;
public function main() {
    xml<xml:Element|xml:Text> a = xml `<foo/>`;
    string b = "foo";
    ("foo"|"bar") c = "foo";
    Foo x2 = a + c; 
}
```
This example gives error as
```
incompatible types: expected 'Foo', found 'xml<(xml:Element|xml:Text|xml:Text)>
```